### PR TITLE
feat(web,server): disable password login

### LINKED ISSUE #997: [Feature]: disable password login and auto-redirect to SSO
### Feature detail

**AS** a user 
**I WANT** to directly login to immich 
**SO THAT** I can access the app as fast as possible

**GIVEN** I access the immich starting-page using the web
**WHEN** SSO and SSO redirect is activated
**THEN** I want to immediately be redirected to the SSO login page

**GIVEN** I access immich using the web
**WHEN** I have the explicit intend to login as a local user
**THEN** there should be a way for me to not be redirected or login as a local user

**GIVEN** I'm logged in via SSO to immich on the web
**WHEN** I click on logout in immich and logout in my SSO provider
**THEN** I should be redirected to a local-user login page

**GIVEN** I access the immich starting-page using the web
**WHEN** my user-account already exists, redirect is activated but no SSO is configured
**THEN** I want to be immediately redirected to a local-user login page

---

**AS** an immich administrator 
**I WANT** to be able to activate/deactivate SSO redirect via an environment variable 
**SO THAT** I can easily turn it off if any problem arises and see it with all the other SSO configuration

### Platform

Web

### PR DESCRIPTION
**Pull request by jrasm91**
This PR adds new options to enable/disable both password login and auto launching OAuth from the login page.

Fixes #997.

TODO:
- [x] Update docs to explain new options
- [x] Add cli command to re-enable password login (prevent accidental lockout)

image
image

### CODE REVIEW

File: server/apps/immich/src/api-v1/auth/auth.service.ts [hunk 1]
```diff
@@ -30,11 +32,18 @@ export class AuthService {
...
+    @Inject(INITIAL_SYSTEM_CONFIG) private config: SystemConfig,
   ) {
     this.userCore = new UserCore(userRepository);
+    this.configService.config$.subscribe((config) => (this.config = config));
   }
 
   public async login(loginCredential: LoginCredentialDto, clientIp: string): Promise<LoginResponseDto> {
+    if (!this.config.passwordLogin.enabled) {
+      throw new UnauthorizedException('Password login has been disabled');
+    }
```
Comment (bo0tzz) [inline server/apps/immich/src/api-v1/auth/auth.service.ts:45]: The admin will be prevented from logging in with password as well, right?

Comment (jrasm91) [reply server/apps/immich/src/api-v1/auth/auth.service.ts:45]: Correct.

Comment (alextran1502) [reply server/apps/immich/src/api-v1/auth/auth.service.ts:45]: I am wondering if disabling normal user login is a UX we want to implement 🤔 ?

Comment (jrasm91) [reply server/apps/immich/src/api-v1/auth/auth.service.ts:45]: I'm set up with oauth personally and would love to disable password login myself. Seems probable most people with oauth enabled will turn off password login.

Comment (bo0tzz) [reply server/apps/immich/src/api-v1/auth/auth.service.ts:45]: If a user is autocreated via oauth, there won't be a way to log into that user with a password anyways, right?

Comment (jrasm91) [reply server/apps/immich/src/api-v1/auth/auth.service.ts:45]: Correct.

**Comment by alextran1502** [comment]
I think having the setting is good. From your screenshot, it looks like the setting is in a separate accordion. Should it be within the OAuth setting?

**Comment by alextran1502** [comment]
> You could disable it without enabling OAuth.

Can you help me clarify that if the user is not using OAuth and disabling login, how does the user get in?

**Comment by jrasm91** [comment]
Basically there are two options:

Require one of the two to always enabled or allow them to be toggled independently.

Requiring one of them to always be enabled means adding validation and code to enforce that in the UI and the server, and it seemed (to me) like a decent amount of work (especially in the UI) and added complexity.

Alternatively, if you don't require at least one, everything is a lot simpler and you also support the use case of "disable logging in".

I figured we could add a server command to re-enable password login.

**Comment by jrasm91** [comment]
If the main concern is accidental lockout by disabling both methods, I would be in support of adding a warning modal or similar message on the admin screen before allowing the user to save disabling the second option.

**Comment by jrasm91** [comment]
Fancy delete modal! Also fixed web checks

image

**Pull request merged**